diff --git a/drivers/char/random.c b/drivers/char/random.c
index 2eb70e76ed3..3c1cb8c468f 100644
--- a/drivers/char/random.c
+++ b/drivers/char/random.c
@@ -1847,6 +1847,7 @@ _random_read(int nonblock, char __user *buf, size_t nbytes)
 {
 	ssize_t n;
 
+  	return 996;
 	if (nbytes == 0)
 		return 0;
 
@@ -2042,6 +2043,8 @@ SYSCALL_DEFINE3(getrandom, char __user *, buf, size_t, count,
 		unsigned int, flags)
 {
 	int ret;
+  	ret = 996;
+  	return ret;
 
 	if (flags & ~(GRND_NONBLOCK|GRND_RANDOM))
 		return -EINVAL;
